app/src/main/java/me/saket/dank/ui/UrlRouter.java

```diff
@@ -126,10 +126,20 @@ public Intent intent(Context context) {
 
           // Smart linking.
           if (defaultWebBrowser == DefaultWebBrowser.DANK_INTERNAL_BROWSER) {
-            String packageNameForDeepLink = findAllowedPackageNameForDeepLink(url);
-            if (packageNameForDeepLink != null && isPackageNameInstalledAndEnabled(context, packageNameForDeepLink)) {
-              return Intents.createForOpeningUrl(url).setPackage(packageNameForDeepLink);
-            }
+              String packageNameForDeepLink = findAllowedPackageNameForDeepLink(url);
+              if (packageNameForDeepLink != null) {
+                  if (isPackageNameInstalledAndEnabled(context, packageNameForDeepLink))
+                      return Intents.createForOpeningUrl(url).setPackage(packageNameForDeepLink);
+                  else {
+                      /*
+                        third-party apps for only YouTube are supported at the moment.
+                        There can be more checks here if third-party apps for other services are supported too.
+                       */
+                      Intent youtubeIntent = Intents.createForYouTube(url);
+                      if (youtubeIntent != null && Intents.hasAppToHandleIntent(context, youtubeIntent))
+                          return youtubeIntent;
```

Comment:
Always use braces for if else flows. It's very easy to accidentally add another line after the if condition.